src/ImageSharp/Formats/Jpeg/Components/Encoder/LuminanceForwardConverter{TPixel}.cs

```diff
@@ -74,6 +79,44 @@ public void Convert(int x, int y, ref RowOctet<TPixel> currentRows)
             ref Block8x8F yBlock = ref this.Y;
             ref L8 l8Start = ref MemoryMarshal.GetReference(this.l8Span);
 
+            if (RgbToYCbCrConverterVectorized.IsSupported)
+            {
+                ConvertAvx(ref l8Start, ref yBlock);
+            }
+            else
+            {
+                ConvertScalar(ref l8Start, ref yBlock);
+            }
+        }
+
+        /// <summary>
+        /// Converts 8x8 L8 pixel matrix to 8x8 Block of floats using Avx2 Intrinsics.
+        /// </summary>
+        /// <param name="l8Start">Start of span of L8 pixels with size of 64</param>
+        /// <param name="yBlock">8x8 destination matrix of Luminance(Y) converted data</param>
+        private static void ConvertAvx(ref L8 l8Start, ref Block8x8F yBlock)
+        {
+            Debug.Assert(RgbToYCbCrConverterVectorized.IsSupported, "AVX2 is required to run this converter");
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+            ref Vector128<byte> l8ByteSpan = ref Unsafe.As<L8, Vector128<byte>>(ref l8Start);
+            ref Vector256<float> destRef = ref yBlock.V0;
+
+            const int bytesPerL8Stride = 8;
+            for (int i = 0; i < 8; i++)
```

Comment:
```suggestion
            for (nint i = 0; i < bytesPerL8Stride; i++)
```
Is `8` from the condition by coincidence the same as `bytesPerL8Stride`? If not, just use `8`, but change to `nint`.

Reply:
Actually it is not. 8 is the number of floats in the destination Vector256. Using the `nint` type does indeed eliminate the cast to IntPtr.
Thanks for the feedback.

Reply:
Thanks @gfoidl for the reviews, always welcome!